### ISSUE TITLE
Error compiling PyTorch model

### ISSUE THREAD

**Issue by andreadasilvabaudet** [aws-neuron/aws-neuron-sdk#662]
Hi team,

I'm trying to compile a PyTorch model and I get the following error: 
```04/19/2023 11:43:43 PM ERROR 497 [neuron-cc]: ***************************************************************
04/19/2023 11:43:43 PM ERROR 497 [neuron-cc]:  An Internal Compiler Error has occurred
04/19/2023 11:43:43 PM ERROR 497 [neuron-cc]: ***************************************************************
04/19/2023 11:43:43 PM ERROR 497 [neuron-cc]: 
04/19/2023 11:43:43 PM ERROR 497 [neuron-cc]: Error message:  
04/19/2023 11:43:43 PM ERROR 497 [neuron-cc]: 
04/19/2023 11:43:43 PM ERROR 497 [neuron-cc]: Error class:    AssertionError
04/19/2023 11:43:43 PM ERROR 497 [neuron-cc]: Error location: Unknown
04/19/2023 11:43:43 PM ERROR 497 [neuron-cc]: Command line:   /opt/conda/bin/neuron-cc compile /home/ec2-user/neuron-compilation/Wav2Lip/workdir/0/graph_def.pb --framework TENSORFLOW --pipeline compile SaveTemps --output /home/ec2-user/neuron-compilation/Wav2Lip/workdir/0/graph_def.neff --io-config '{"inputs": {"0:0": [[128, 1, 80, 16], "float32"], "1:0": [[128, 6, 96, 96], "float32"]}, "outputs": ["Sequential_66/Sigmoid_5/aten_sigmoid/Sigmoid:0"]}' --verbose 35
04/19/2023 11:43:43 PM ERROR 497 [neuron-cc]: 
04/19/2023 11:43:43 PM ERROR 497 [neuron-cc]: Internal details:
04/19/2023 11:43:43 PM ERROR 497 [neuron-cc]:   File "neuroncc/driver/CommandDriver.py", line 224, in neuroncc.driver.CommandDriver.CommandDriver.run
04/19/2023 11:43:43 PM ERROR 497 [neuron-cc]:   File "neuroncc/driver/commands/CompileCommand.py", line 580, in neuroncc.driver.commands.CompileCommand.CompileCommand.run
04/19/2023 11:43:43 PM ERROR 497 [neuron-cc]:   File "neuroncc/driver/commands/CompileCommand.py", line 558, in neuroncc.driver.commands.CompileCommand.CompileCommand.runPipeline
04/19/2023 11:43:43 PM ERROR 497 [neuron-cc]:   File "neuroncc/driver/commands/CompileCommand.py", line 562, in neuroncc.driver.commands.CompileCommand.CompileCommand.runPipeline
04/19/2023 11:43:43 PM ERROR 497 [neuron-cc]:   File "neuroncc/driver/Job.py", line 289, in neuroncc.driver.Job.SingleInputJob.run
04/19/2023 11:43:43 PM ERROR 497 [neuron-cc]:   File "neuroncc/driver/Pipeline.py", line 30, in neuroncc.driver.Pipeline.Pipeline.runSingleInput
04/19/2023 11:43:43 PM ERROR 497 [neuron-cc]:   File "neuroncc/driver/Job.py", line 289, in neuroncc.driver.Job.SingleInputJob.run
04/19/2023 11:43:43 PM ERROR 497 [neuron-cc]:   File "neuroncc/driver/Pipeline.py", line 30, in neuroncc.driver.Pipeline.Pipeline.runSingleInput
04/19/2023 11:43:43 PM ERROR 497 [neuron-cc]:   File "neuroncc/driver/Job.py", line 289, in neuroncc.driver.Job.SingleInputJob.run
04/19/2023 11:43:43 PM ERROR 497 [neuron-cc]:   File "neuroncc/driver/jobs/Frontend.py", line 429, in neuroncc.driver.jobs.Frontend.Frontend.runSingleInput
04/19/2023 11:43:43 PM ERROR 497 [neuron-cc]:   File "neuroncc/driver/jobs/Frontend.py", line 379, in neuroncc.driver.jobs.Frontend.Frontend.runTVMFrontend
04/19/2023 11:43:43 PM ERROR 497 [neuron-cc]:   File "neuroncc/driver/jobs/Frontend.py", line 380, in neuroncc.driver.jobs.Frontend.Frontend.runTVMFrontend
04/19/2023 11:43:43 PM ERROR 497 [neuron-cc]:   File "neuroncc/driver/jobs/Frontend.py", line 390, in neuroncc.driver.jobs.Frontend.Frontend.runTVMFrontend
04/19/2023 11:43:43 PM ERROR 497 [neuron-cc]:   File "neuroncc/driver/jobs/Frontend.py", line 360, in neuroncc.driver.jobs.Frontend.Frontend.runTVMFrontend.tensorize
04/19/2023 11:43:43 PM ERROR 497 [neuron-cc]:   File "neuroncc/starfish/penguin/Frontend.py", line 47, in neuroncc.starfish.penguin.Frontend.tensorizeRelay
04/19/2023 11:43:43 PM ERROR 497 [neuron-cc]:   File "neuroncc/starfish/penguin/Frontend.py", line 48, in neuroncc.starfish.penguin.Frontend.tensorizeRelay
04/19/2023 11:43:43 PM ERROR 497 [neuron-cc]:   File "neuroncc/starfish/penguin/Frontend.py", line 69, in neuroncc.starfish.penguin.Frontend.tensorizeRelay
04/19/2023 11:43:43 PM ERROR 497 [neuron-cc]:   File "neuroncc/starfish/penguin/Compile.py", line 262, in neuroncc.starfish.penguin.Compile.compile_cu
04/19/2023 11:43:43 PM ERROR 497 [neuron-cc]:   File "neuroncc/starfish/penguin/DotTransform.py", line 378, in neuroncc.starfish.penguin.DotTransform.PassManager.transformFunction
04/19/2023 11:43:43 PM ERROR 497 [neuron-cc]:   File "neuroncc/starfish/penguin/DotTransform.py", line 124, in neuroncc.starfish.penguin.DotTransform.DotTransform.runOnFunction
04/19/2023 11:43:43 PM ERROR 497 [neuron-cc]:   File "neuroncc/starfish/penguin/DotTransform.py", line 178, in neuroncc.starfish.penguin.DotTransform.DotTransform.run_with_exception_handling
04/19/2023 11:43:43 PM ERROR 497 [neuron-cc]:   File "neuroncc/starfish/penguin/DotTransform.py", line 160, in neuroncc.starfish.penguin.DotTransform.DotTransform.run_with_exception_handling
04/19/2023 11:43:43 PM ERROR 497 [neuron-cc]:   File "neuroncc/starfish/penguin/DotTransform.py", line 186, in neuroncc.starfish.penguin.DotTransform.DotTransform.timed_run_
04/19/2023 11:43:43 PM ERROR 497 [neuron-cc]:   File "neuroncc/starfish/penguin/DotTransform.py", line 188, in neuroncc.starfish.penguin.DotTransform.DotTransform.timed_run_
04/19/2023 11:43:43 PM ERROR 497 [neuron-cc]:   File "neuroncc/starfish/penguin/DotTransform.py", line 189, in neuroncc.starfish.penguin.DotTransform.DotTransform.timed_run_
04/19/2023 11:43:43 PM ERROR 497 [neuron-cc]:   File "neuroncc/starfish/penguin/DotTransform.py", line 206, in neuroncc.starfish.penguin.DotTransform.DotTransform.run_
04/19/2023 11:43:43 PM ERROR 497 [neuron-cc]:   File "neuroncc/starfish/penguin/DotTransform.py", line 207, in neuroncc.starfish.penguin.DotTransform.DotTransform.run_
04/19/2023 11:43:43 PM ERROR 497 [neuron-cc]:   File "neuroncc/starfish/penguin/DotTransform.py", line 287, in neuroncc.starfish.penguin.DotTransform.DotTransform.transformFunction
04/19/2023 11:43:43 PM ERROR 497 [neuron-cc]:   File "neuroncc/starfish/penguin/DotTransform.py", line 281, in neuroncc.starfish.penguin.DotTransform.DotTransform.runTransforms
04/19/2023 11:43:43 PM ERROR 497 [neuron-cc]:   File "neuroncc/starfish/penguin/targets/tonga/passes/InferInitValue.py", line 96, in neuroncc.starfish.penguin.targets.tonga.passes.InferInitValue.InferInitValue.transformStmts
04/19/2023 11:43:43 PM ERROR 497 [neuron-cc]:   File "neuroncc/starfish/penguin/DotTransform.py", line 115, in neuroncc.starfish.penguin.DotTransform.DotTransform.transform
04/19/2023 11:43:43 PM ERROR 497 [neuron-cc]:   File "neuroncc/starfish/penguin/targets/tonga/passes/InferInitValue.py", line 67, in neuroncc.starfish.penguin.targets.tonga.passes.InferInitValue.InferInitValue.transformTensor
04/19/2023 11:43:43 PM ERROR 497 [neuron-cc]:   File "neuroncc/starfish/penguin/targets/tonga/passes/InferInitValue.py", line 21, in neuroncc.starfish.penguin.targets.tonga.passes.InferInitValue.store_access_ranges
04/19/2023 11:43:43 PM ERROR 497 [neuron-cc]:   File "neuroncc/starfish/penguin/ir/AffinePredicate.py", line 172, in neuroncc.starfish.penguin.ir.AffinePredicate.AffinePredicate.project
04/19/2023 11:43:43 PM ERROR 497 [neuron-cc]: 
04/19/2023 11:43:43 PM ERROR 497 [neuron-cc]: Version information:
04/19/2023 11:43:43 PM ERROR 497 [neuron-cc]:   Neuron Compiler version 1.14.3.0+adaa2ac56
04/19/2023 11:43:43 PM ERROR 497 [neuron-cc]:   
04/19/2023 11:43:43 PM ERROR 497 [neuron-cc]:   HWM version 1.14.0.0-a9fb5c73a
04/19/2023 11:43:43 PM ERROR 497 [neuron-cc]:   NEFF version Dynamic
04/19/2023 11:43:43 PM ERROR 497 [neuron-cc]:   TVM version 1.14.0.0+0
04/19/2023 11:43:43 PM ERROR 497 [neuron-cc]:   NumPy version 1.18.5
04/19/2023 11:43:43 PM ERROR 497 [neuron-cc]:   MXNet not available
04/19/2023 11:43:43 PM ERROR 497 [neuron-cc]:   TF not available
04/19/2023 11:43:43 PM ERROR 497 [neuron-cc]: 
04/19/2023 11:43:43 PM ERROR 497 [neuron-cc]: Artifacts stored in: /home/ec2-user/neuron-compilation/Wav2Lip/workdir/0
```

Here is my script for compilation:
```import torch
import torch_neuron
image = (torch.zeros([128, 1, 80, 16], dtype=torch.float32), torch.zeros([128, 6, 96, 96], dtype=torch.float32))
model_neuron = torch.neuron.trace(model, example_inputs=image, compiler_workdir='./workdir')
model_neuron.save("model.pt")
```

These are the dependencies installed:
- aws-neuronx-dkms.noarch            2.8.4.0-dkms                 @neuron
- aws-neuronx-tools.x86_64            2.9.5.0-1                        @neuron
- neuron-cc               1.14.3.0+adaa2ac56
- torch-neuron           1.13.1.2.6.6.0
- Python version: Python 3.7.16

This code is executed using an EC2 instance inf1.2xlarge using the Deep Learning AMI GPU PyTorch 1.12.1 (Amazon Linux 2) 20230324.

Thank you in advance.

**Comment by micwade-aws** [comment]
@andreadasilvabaudet Thanks for reporting. Possible to send us the log from above'04/19/2023 11:43:43 PM ERROR 497 [neuron-cc]:  An Internal Compiler Error has occurred'?  

There may be clues in the log that can help us unblock you, such as a specific failing operator.

**Comment by andreadasilvabaudet** [comment]
Hi @micwade-aws,

Here are the previous logs:

```05/03/2023 07:31:46 PM INFO 29116 [root]: Intermediate files stored in /home/ec2-user/neuron-compilation/Wav2Lip/workdir_aws/0, output in /home/ec2-user/neuron-compilation/Wav2Lip/workdir_aws/0
05/03/2023 07:31:46 PM INFO 29116 [pipeline.custom.0]: Running pipeline custom.0
05/03/2023 07:31:46 PM INFO 29116 [pipeline.custom.0]: Starting job pipeline.compile.0 state state 0
05/03/2023 07:31:46 PM INFO 29116 [pipeline.compile.0]: Running pipeline compile.0
05/03/2023 07:31:46 PM INFO 29116 [pipeline.compile.0]: Starting job job.Frontend.4 state state 0
05/03/2023 07:31:46 PM INFO 29116 [job.Frontend.4]: Replay this job by calling: /home/ec2-user/aws_neuron_venv_pytorch/bin/neuron-cc compile --framework TENSORFLOW --state '{"model": ["/home/ec2-user/neuron-compilation/Wav2Lip/workdir_aws/0/graph_def.pb"], "state_dir": "/home/ec2-user/neuron-compilation/Wav2Lip/workdir_aws/0", "state_id": "root"}' --pipeline Frontend --framework TENSORFLOW --io-config '{"inputs": {"0:0": [[128, 1, 80, 16], "float32"], "1:0": [[128, 6, 96, 96], "float32"]}, "outputs": ["Sequential_66/Sigmoid_5/aten_sigmoid/Sigmoid:0"]}'
05/03/2023 07:31:51 PM INFO 29116 [job.Frontend.4]: IR signature: 169786284c0bfb7a65d161ee09a5b5e7d0baca7ccdfa46c864a8d8c53c69791c for graph_def.pb
05/03/2023 07:31:51 PM INFO 29116 [job.Frontend.4]: total padded opcount is 826804273152
05/03/2023 07:31:51 PM INFO 29116 [job.Frontend.4]: Finished Jellyfish import
05/03/2023 07:31:51 PM INFO 29116 [job.Frontend.4]: Start tensorization
05/03/2023 07:31:58 PM INFO 29116 [job.Frontend.4]: IR signature: 25f776740ea206ce9cca649a6bff2fb3043c22f090d3f13fa602fee20bc0f804 for relay_graph_post_opt_unit_level.txt
05/03/2023 07:31:58 PM INFO 29116 [root/Tensorizer/All]: Enter time region
05/03/2023 07:31:58 PM INFO 29116 [Tensorizer]: Tensorizer options:  --fp32-cast=all  --disable-bitcasted-transpose  --disable-expensive-checks  --disable-max-stride-tiling  --enable-replication  --max-local-tensor-tile-size-in-bytes=16384  --tensor-layout-p-order=0  --tensor-layout-b-order=1  --enable-advanced-delinearization  --weight-coalescing-threshold=512  --disable-shifted-fusion  --enable-bir-converter=enable 
05/03/2023 07:31:58 PM INFO 29116 [sg00/Tensorizer/RelayPasses]: Enter time region
05/03/2023 07:31:59 PM INFO 29116 [sg00/Tensorizer/RelayPasses]: Exit time region: delta=0.687s
05/03/2023 07:31:59 PM INFO 29116 [sg00/Tensorizer/IRBuilder]: Enter time region
05/03/2023 07:32:00 PM INFO 29116 [sg00/Tensorizer/IRBuilder]: Exit time region: delta=0.852s
05/03/2023 07:32:00 PM INFO 29116 [Tensorizer]: Tensorizer options:  --fp32-cast=all  --disable-bitcasted-transpose  --disable-expensive-checks  --disable-max-stride-tiling  --enable-replication  --max-local-tensor-tile-size-in-bytes=16384  --tensor-layout-p-order=0  --tensor-layout-b-order=1  --enable-advanced-delinearization  --weight-coalescing-threshold=512  --disable-shifted-fusion  --enable-bir-converter=enable 
05/03/2023 07:32:00 PM INFO 29116 [Tensorizer]: Generate: 1 statements from relay. Start optimizations ...
05/03/2023 07:32:00 PM INFO 29116 [Statistics]: Weights total number of bytes: 145284300
05/03/2023 07:32:00 PM INFO 29116 [Statistics]: RelayIF total number of bytes: 34267136.0
05/03/2023 07:32:00 PM INFO 29116 [Statistics]: RelayOF total number of bytes: 14155776.0
05/03/2023 07:32:00 PM INFO 29116 [Statistics]: Weights total number of bytes: 145284300.0
05/03/2023 07:32:00 PM INFO 29116 [sg00/Tensorizer/DoNothing]: Enter time region
05/03/2023 07:32:00 PM INFO 29116 [DoNothing]: Finished (changed=False)
05/03/2023 07:32:00 PM INFO 29116 [sg00/Tensorizer/DoNothing]: Exit time region: delta=0.180s
05/03/2023 07:32:00 PM INFO 29116 [sg00/Tensorizer/MutateDataType]: Enter time region
05/03/2023 07:32:00 PM INFO 29116 [MutateDataType]: Finished (changed=False)
05/03/2023 07:32:00 PM INFO 29116 [sg00/Tensorizer/MutateDataType]: Exit time region: delta=0.179s
05/03/2023 07:32:00 PM INFO 29116 [sg00/Tensorizer/ResolveEqualPredicates]: Enter time region
05/03/2023 07:32:01 PM INFO 29116 [ResolveEqualPredicates]: Finished (changed=False)
05/03/2023 07:32:01 PM INFO 29116 [sg00/Tensorizer/ResolveEqualPredicates]: Exit time region: delta=0.179s
05/03/2023 07:32:01 PM INFO 29116 [sg00/Tensorizer/EliminateDivs]: Enter time region
05/03/2023 07:32:01 PM INFO 29116 [EliminateDivs]: Finished (changed=False)
05/03/2023 07:32:01 PM INFO 29116 [sg00/Tensorizer/EliminateDivs]: Exit time region: delta=0.180s
05/03/2023 07:32:01 PM INFO 29116 [sg00/Tensorizer/SimplifyPredicates]: Enter time region
05/03/2023 07:32:01 PM INFO 29116 [SimplifyPredicates]: Finished (changed=True #instances=2857176924160)
05/03/2023 07:32:02 PM INFO 29116 [sg00/Tensorizer/SimplifyPredicates]: Exit time region: delta=0.765s
05/03/2023 07:32:02 PM INFO 29116 [sg00/Tensorizer/Simplifier]: Enter time region
05/03/2023 07:32:02 PM INFO 29116 [Simplifier]: Finished (changed=False)
05/03/2023 07:32:02 PM INFO 29116 [sg00/Tensorizer/Simplifier]: Exit time region: delta=0.275s
05/03/2023 07:32:02 PM INFO 29116 [sg00/Tensorizer/TCTransform]: Enter time region
05/03/2023 07:32:02 PM INFO 29116 [TCTransform]: Finished (changed=True #instances=2289847238656)
05/03/2023 07:32:02 PM INFO 29116 [sg00/Tensorizer/TCTransform]: Exit time region: delta=0.210s
05/03/2023 07:32:02 PM INFO 29116 [sg00/Tensorizer/CommuteConcat]: Enter time region
05/03/2023 07:32:02 PM INFO 29116 [CommuteConcat]: Finished (changed=False)
05/03/2023 07:32:02 PM INFO 29116 [sg00/Tensorizer/CommuteConcat]: Exit time region: delta=0.183s
05/03/2023 07:32:02 PM INFO 29116 [sg00/Tensorizer/LoopFusion]: Enter time region
05/03/2023 07:32:05 PM INFO 29116 [LoopFusion]: Finished (changed=True #instances=7771200716800)
05/03/2023 07:32:05 PM INFO 29116 [sg00/Tensorizer/LoopFusion]: Exit time region: delta=2.437s
05/03/2023 07:32:05 PM INFO 29116 [sg00/Tensorizer/Simplifier]: Enter time region
05/03/2023 07:32:05 PM INFO 29116 [Simplifier]: Finished (changed=True #instances=6944612450304)
05/03/2023 07:32:05 PM INFO 29116 [sg00/Tensorizer/Simplifier]: Exit time region: delta=0.515s
05/03/2023 07:32:05 PM INFO 29116 [sg00/Tensorizer/DelinearIndices]: Enter time region
05/03/2023 07:32:05 PM INFO 29116 [DelinearIndices]: Finished (changed=True #instances=6944612450304)
05/03/2023 07:32:05 PM INFO 29116 [sg00/Tensorizer/DelinearIndices]: Exit time region: delta=0.237s
05/03/2023 07:32:05 PM INFO 29116 [sg00/Tensorizer/Delinearization]: Enter time region
05/03/2023 07:32:06 PM INFO 29116 [Delinearization]: Finished (changed=False)
05/03/2023 07:32:06 PM INFO 29116 [sg00/Tensorizer/Delinearization]: Exit time region: delta=0.087s
05/03/2023 07:32:06 PM INFO 29116 [sg00/Tensorizer/DeadStoreElimination]: Enter time region
05/03/2023 07:32:07 PM INFO 29116 [DeadStoreElimination]: Finished (changed=True #instances=6722992988160)
05/03/2023 07:32:07 PM INFO 29116 [sg00/Tensorizer/DeadStoreElimination]: Exit time region: delta=1.832s
05/03/2023 07:32:07 PM INFO 29116 [sg00/Tensorizer/Simplifier]: Enter time region
05/03/2023 07:32:08 PM INFO 29116 [Simplifier]: Finished (changed=False)
05/03/2023 07:32:08 PM INFO 29116 [sg00/Tensorizer/Simplifier]: Exit time region: delta=0.226s
05/03/2023 07:32:08 PM INFO 29116 [sg00/Tensorizer/LICM]: Enter time region
05/03/2023 07:32:08 PM INFO 29116 [LICM]: Finished (changed=True #instances=1544918843392)
05/03/2023 07:32:08 PM INFO 29116 [sg00/Tensorizer/LICM]: Exit time region: delta=0.072s
05/03/2023 07:32:08 PM INFO 29116 [sg00/Tensorizer/DelinearIndices]: Enter time region
05/03/2023 07:32:08 PM INFO 29116 [DelinearIndices]: Finished (changed=False)
05/03/2023 07:32:08 PM INFO 29116 [sg00/Tensorizer/DelinearIndices]: Exit time region: delta=0.077s
05/03/2023 07:32:08 PM INFO 29116 [sg00/Tensorizer/Delinearization]: Enter time region
05/03/2023 07:32:08 PM INFO 29116 [Delinearization]: Finished (changed=False)
05/03/2023 07:32:08 PM INFO 29116 [sg00/Tensorizer/Delinearization]: Exit time region: delta=0.084s
05/03/2023 07:32:08 PM INFO 29116 [sg00/Tensorizer/LoopFusion]: Enter time region
05/03/2023 07:32:08 PM INFO 29116 [LoopFusion]: Finished (changed=True #instances=1544607612928)
05/03/2023 07:32:08 PM INFO 29116 [sg00/Tensorizer/LoopFusion]: Exit time region: delta=0.455s
05/03/2023 07:32:08 PM INFO 29116 [sg00/Tensorizer/LICM]: Enter time region
05/03/2023 07:32:08 PM INFO 29116 [LICM]: Finished (changed=False)
05/03/2023 07:32:08 PM INFO 29116 [sg00/Tensorizer/LICM]: Exit time region: delta=0.049s
05/03/2023 07:32:08 PM INFO 29116 [sg00/Tensorizer/Simplifier]: Enter time region
05/03/2023 07:32:09 PM INFO 29116 [Simplifier]: Finished (changed=False)
05/03/2023 07:32:09 PM INFO 29116 [sg00/Tensorizer/Simplifier]: Exit time region: delta=0.193s
05/03/2023 07:32:09 PM INFO 29116 [sg00/Tensorizer/ValueNumbering]: Enter time region
05/03/2023 07:32:09 PM INFO 29116 [ValueNumbering]: Finished (changed=True #instances=1544263581696)
05/03/2023 07:32:09 PM INFO 29116 [sg00/Tensorizer/ValueNumbering]: Exit time region: delta=0.116s
05/03/2023 07:32:09 PM INFO 29116 [sg00/Tensorizer/LICM]: Enter time region
05/03/2023 07:32:09 PM INFO 29116 [LICM]: Finished (changed=False)
05/03/2023 07:32:09 PM INFO 29116 [sg00/Tensorizer/LICM]: Exit time region: delta=0.046s
05/03/2023 07:32:09 PM INFO 29116 [sg00/Tensorizer/MemcpyElimination]: Enter time region
05/03/2023 07:32:09 PM INFO 29116 [MemcpyElimination]: Finished (changed=True #instances=1544078057472)
05/03/2023 07:32:09 PM INFO 29116 [sg00/Tensorizer/MemcpyElimination]: Exit time region: delta=0.260s
05/03/2023 07:32:09 PM INFO 29116 [sg00/Tensorizer/PadElimination]: Enter time region
05/03/2023 07:32:09 PM INFO 29116 [PadElimination]: Finished (changed=False)
05/03/2023 07:32:09 PM INFO 29116 [sg00/Tensorizer/PadElimination]: Exit time region: delta=0.041s
05/03/2023 07:32:09 PM INFO 29116 [sg00/Tensorizer/DelinearIndices]: Enter time region
05/03/2023 07:32:09 PM INFO 29116 [DelinearIndices]: Finished (changed=True #instances=1544078057472)
05/03/2023 07:32:09 PM INFO 29116 [sg00/Tensorizer/DelinearIndices]: Exit time region: delta=0.095s
05/03/2023 07:32:09 PM INFO 29116 [sg00/Tensorizer/Delinearization]: Enter time region
05/03/2023 07:32:09 PM INFO 29116 [Delinearization]: Finished (changed=False)
05/03/2023 07:32:09 PM INFO 29116 [sg00/Tensorizer/Delinearization]: Exit time region: delta=0.069s
05/03/2023 07:32:09 PM INFO 29116 [sg00/Tensorizer/LoopFusion]: Enter time region
05/03/2023 07:32:10 PM INFO 29116 [LoopFusion]: Finished (changed=True #instances=1544067571712)
05/03/2023 07:32:10 PM INFO 29116 [sg00/Tensorizer/LoopFusion]: Exit time region: delta=0.416s
05/03/2023 07:32:10 PM INFO 29116 [sg00/Tensorizer/Simplifier]: Enter time region
05/03/2023 07:32:10 PM INFO 29116 [Simplifier]: Finished (changed=False)
05/03/2023 07:32:10 PM INFO 29116 [sg00/Tensorizer/Simplifier]: Exit time region: delta=0.187s
05/03/2023 07:32:10 PM INFO 29116 [sg00/Tensorizer/LICM]: Enter time region
05/03/2023 07:32:10 PM INFO 29116 [LICM]: Finished (changed=False)
05/03/2023 07:32:10 PM INFO 29116 [sg00/Tensorizer/LICM]: Exit time region: delta=0.044s
05/03/2023 07:32:10 PM INFO 29116 [sg00/Tensorizer/ValueNumbering]: Enter time region
05/03/2023 07:32:10 PM INFO 29116 [ValueNumbering]: Finished (changed=False)
05/03/2023 07:32:10 PM INFO 29116 [sg00/Tensorizer/ValueNumbering]: Exit time region: delta=0.069s
05/03/2023 07:32:10 PM INFO 29116 [sg00/Tensorizer/TCTransform]: Enter time region
05/03/2023 07:32:10 PM INFO 29116 [TCTransform]: Finished (changed=False)
05/03/2023 07:32:10 PM INFO 29116 [sg00/Tensorizer/TCTransform]: Exit time region: delta=0.045s
05/03/2023 07:32:10 PM INFO 29116 [sg00/Tensorizer/CommuteConcat]: Enter time region
05/03/2023 07:32:10 PM INFO 29116 [CommuteConcat]: Finished (changed=True #instances=1544448270336)
05/03/2023 07:32:10 PM INFO 29116 [sg00/Tensorizer/CommuteConcat]: Exit time region: delta=0.237s
05/03/2023 07:32:10 PM INFO 29116 [sg00/Tensorizer/LoopFusion]: Enter time region
05/03/2023 07:32:11 PM INFO 29116 [LoopFusion]: Finished (changed=True #instances=1544182587392)
05/03/2023 07:32:11 PM INFO 29116 [sg00/Tensorizer/LoopFusion]: Exit time region: delta=0.442s
05/03/2023 07:32:11 PM INFO 29116 [sg00/Tensorizer/ValueNumbering]: Enter time region
05/03/2023 07:32:11 PM INFO 29116 [ValueNumbering]: Finished (changed=False)
05/03/2023 07:32:11 PM INFO 29116 [sg00/Tensorizer/ValueNumbering]: Exit time region: delta=0.068s
05/03/2023 07:32:11 PM INFO 29116 [sg00/Tensorizer/RecognizeOpIdiom]: Enter time region
05/03/2023 07:32:11 PM INFO 29116 [RecognizeOpIdiom]: Finished (changed=False)
05/03/2023 07:32:11 PM INFO 29116 [sg00/Tensorizer/RecognizeOpIdiom]: Exit time region: delta=0.067s
05/03/2023 07:32:11 PM INFO 29116 [sg00/Tensorizer/DeadCodeElimination]: Enter time region
05/03/2023 07:32:11 PM INFO 29116 [DeadCodeElimination]: Finished (changed=False)
05/03/2023 07:32:11 PM INFO 29116 [sg00/Tensorizer/DeadCodeElimination]: Exit time region: delta=0.048s
05/03/2023 07:32:11 PM INFO 29116 [Tensorizer]: After optimization: 1 statements
05/03/2023 07:32:11 PM INFO 29116 [sg00/Tensorizer/AutoCastFP32]: Enter time region
05/03/2023 07:32:11 PM INFO 29116 [AutoCastFP32]: Finished (changed=True #instances=1549829562368)
05/03/2023 07:32:11 PM INFO 29116 [sg00/Tensorizer/AutoCastFP32]: Exit time region: delta=0.059s
05/03/2023 07:32:11 PM INFO 29116 [sg00/Tensorizer/LoopFusion]: Enter time region
05/03/2023 07:32:11 PM INFO 29116 [LoopFusion]: Finished (changed=False)
05/03/2023 07:32:11 PM INFO 29116 [sg00/Tensorizer/LoopFusion]: Exit time region: delta=0.284s
05/03/2023 07:32:11 PM INFO 29116 [sg00/Tensorizer/Simplifier]: Enter time region
05/03/2023 07:32:11 PM INFO 29116 [Simplifier]: Finished (changed=False)
05/03/2023 07:32:11 PM INFO 29116 [sg00/Tensorizer/Simplifier]: Exit time region: delta=0.160s
05/03/2023 07:32:11 PM INFO 29116 [sg00/Tensorizer/Delinearization]: Enter time region
05/03/2023 07:32:11 PM INFO 29116 [Delinearization]: Finished (changed=True #instances=1549829562368)
05/03/2023 07:32:12 PM INFO 29116 [sg00/Tensorizer/Delinearization]: Exit time region: delta=0.082s
05/03/2023 07:32:12 PM INFO 29116 [sg00/Tensorizer/ResolveAccessConflict]: Enter time region
05/03/2023 07:32:12 PM INFO 29116 [ResolveAccessConflict]: Finished (changed=False)
05/03/2023 07:32:12 PM INFO 29116 [sg00/Tensorizer/ResolveAccessConflict]: Exit time region: delta=0.109s
05/03/2023 07:32:12 PM INFO 29116 [sg00/Tensorizer/TransformLayout]: Enter time region
05/03/2023 07:32:12 PM INFO 29116 [TransformLayout]: Finished (changed=True #instances=1549926006784)
05/03/2023 07:32:12 PM INFO 29116 [sg00/Tensorizer/TransformLayout]: Exit time region: delta=0.119s
05/03/2023 07:32:12 PM INFO 29116 [sg00/Tensorizer/PartitionLocalityOpt]: Enter time region
05/03/2023 07:32:12 PM INFO 29116 [PartitionLocalityOpt]: Finished (changed=True #instances=1549926006784)
05/03/2023 07:32:12 PM INFO 29116 [sg00/Tensorizer/PartitionLocalityOpt]: Exit time region: delta=0.098s
05/03/2023 07:32:12 PM INFO 29116 [sg00/Tensorizer/FlattenPartitionAxes]: Enter time region
05/03/2023 07:32:12 PM INFO 29116 [FlattenPartitionAxes]: Finished (changed=True #instances=1549926006784)
05/03/2023 07:32:12 PM INFO 29116 [sg00/Tensorizer/FlattenPartitionAxes]: Exit time region: delta=0.065s
05/03/2023 07:32:12 PM INFO 29116 [sg00/Tensorizer/TongaSizeTiling]: Enter time region
05/03/2023 07:32:13 PM INFO 29116 [TongaSizeTiling]: Finished (changed=True #instances=548535)
05/03/2023 07:32:13 PM INFO 29116 [sg00/Tensorizer/TongaSizeTiling]: Exit time region: delta=1.274s
05/03/2023 07:32:13 PM INFO 29116 [sg00/Tensorizer/TilingProfiler]: Enter time region
05/03/2023 07:32:13 PM INFO 29116 [TilingProfiler]: Finished (changed=False)
05/03/2023 07:32:13 PM INFO 29116 [sg00/Tensorizer/TilingProfiler]: Exit time region: delta=0.076s
05/03/2023 07:32:13 PM INFO 29116 [Tensorizer]: Default layout and tiling instruction count within range.
05/03/2023 07:32:13 PM INFO 29116 [sg00/Tensorizer/FlattenMacroLoop]: Enter time region
05/03/2023 07:32:14 PM INFO 29116 [FlattenMacroLoop]: Finished (changed=True #instances=548535)
05/03/2023 07:32:14 PM INFO 29116 [sg00/Tensorizer/FlattenMacroLoop]: Exit time region: delta=0.163s
05/03/2023 07:32:14 PM INFO 29116 [sg00/Tensorizer/RetileSIMDMacro]: Enter time region
05/03/2023 07:32:14 PM INFO 29116 [RetileSIMDMacro]: Finished (changed=False)
05/03/2023 07:32:14 PM INFO 29116 [sg00/Tensorizer/RetileSIMDMacro]: Exit time region: delta=0.065s
05/03/2023 07:32:14 PM INFO 29116 [sg00/Tensorizer/InferTongaTensor]: Enter time region
05/03/2023 07:32:14 PM INFO 29116 [InferTongaTensor]: Finished (changed=True #instances=548535)
05/03/2023 07:32:14 PM INFO 29116 [sg00/Tensorizer/InferTongaTensor]: Exit time region: delta=0.309s
05/03/2023 07:32:14 PM INFO 29116 [sg00/Tensorizer/TongaSimplifier]: Enter time region
05/03/2023 07:32:14 PM INFO 29116 [TongaSimplifier]: Finished (changed=False)
05/03/2023 07:32:14 PM INFO 29116 [sg00/Tensorizer/TongaSimplifier]: Exit time region: delta=0.089s
05/03/2023 07:32:14 PM INFO 29116 [sg00/Tensorizer/LICM]: Enter time region
05/03/2023 07:32:14 PM INFO 29116 [LICM]: Finished (changed=True #instances=548535)
05/03/2023 07:32:14 PM INFO 29116 [sg00/Tensorizer/LICM]: Exit time region: delta=0.081s
05/03/2023 07:32:14 PM INFO 29116 [sg00/Tensorizer/RewriteReplicationMatmul]: Enter time region
05/03/2023 07:32:14 PM INFO 29116 [RewriteReplicationMatmul]: Finished (changed=True #instances=552119)
05/03/2023 07:32:14 PM INFO 29116 [sg00/Tensorizer/RewriteReplicationMatmul]: Exit time region: delta=0.078s
05/03/2023 07:32:14 PM INFO 29116 [sg00/Tensorizer/FlattenMacroLoop]: Enter time region
05/03/2023 07:32:14 PM INFO 29116 [FlattenMacroLoop]: Finished (changed=True #instances=552119)
05/03/2023 07:32:14 PM INFO 29116 [sg00/Tensorizer/FlattenMacroLoop]: Exit time region: delta=0.142s
05/03/2023 07:32:14 PM INFO 29116 [sg00/Tensorizer/SimplifyPredicates]: Enter time region
05/03/2023 07:32:15 PM INFO 29116 [SimplifyPredicates]: Finished (changed=True #instances=552119)
05/03/2023 07:32:15 PM INFO 29116 [sg00/Tensorizer/SimplifyPredicates]: Exit time region: delta=1.021s
05/03/2023 07:32:15 PM INFO 29116 [sg00/Tensorizer/DataLocalityOpt]: Enter time region
05/03/2023 07:32:21 PM INFO 29116 [DataLocalityOpt]: Finished (changed=True #instances=559875)
05/03/2023 07:32:21 PM INFO 29116 [sg00/Tensorizer/DataLocalityOpt]: Exit time region: delta=5.706s
05/03/2023 07:32:21 PM INFO 29116 [sg00/Tensorizer/TongaSimplifier]: Enter time region
05/03/2023 07:32:21 PM INFO 29116 [TongaSimplifier]: Finished (changed=False)
05/03/2023 07:32:21 PM INFO 29116 [sg00/Tensorizer/TongaSimplifier]: Exit time region: delta=0.171s
05/03/2023 07:32:21 PM INFO 29116 [sg00/Tensorizer/LegalizeTongaMacro]: Enter time region
05/03/2023 07:32:22 PM INFO 29116 [LegalizeTongaMacro]: Finished (changed=True #instances=567387)
05/03/2023 07:32:22 PM INFO 29116 [sg00/Tensorizer/LegalizeTongaMacro]: Exit time region: delta=0.348s
05/03/2023 07:32:22 PM INFO 29116 [sg00/Tensorizer/TongaSimplifier]: Enter time region
05/03/2023 07:32:22 PM INFO 29116 [TongaSimplifier]: Finished (changed=False)
05/03/2023 07:32:22 PM INFO 29116 [sg00/Tensorizer/TongaSimplifier]: Exit time region: delta=0.179s
05/03/2023 07:32:22 PM INFO 29116 [sg00/Tensorizer/PerfectLoopNest]: Enter time region
05/03/2023 07:32:22 PM INFO 29116 [PerfectLoopNest]: Finished (changed=False)
05/03/2023 07:32:22 PM INFO 29116 [sg00/Tensorizer/PerfectLoopNest]: Exit time region: delta=0.161s
05/03/2023 07:32:22 PM INFO 29116 [sg00/Tensorizer/FlattenMacroLoop]: Enter time region
05/03/2023 07:32:22 PM INFO 29116 [FlattenMacroLoop]: Finished (changed=True #instances=567387)
05/03/2023 07:32:22 PM INFO 29116 [sg00/Tensorizer/FlattenMacroLoop]: Exit time region: delta=0.326s
05/03/2023 07:32:22 PM INFO 29116 [sg00/Tensorizer/RewriteWeights]: Enter time region
05/03/2023 07:32:24 PM INFO 29116 [RewriteWeights]: Finished (changed=True #instances=567387)
05/03/2023 07:32:24 PM INFO 29116 [sg00/Tensorizer/RewriteWeights]: Exit time region: delta=1.545s
05/03/2023 07:32:24 PM INFO 29116 [sg00/Tensorizer/ReshapeWeights]: Enter time region
05/03/2023 07:32:24 PM INFO 29116 [ReshapeWeights]: Finished (changed=True #instances=567387)
05/03/2023 07:32:24 PM INFO 29116 [sg00/Tensorizer/ReshapeWeights]: Exit time region: delta=0.209s
05/03/2023 07:32:24 PM INFO 29116 [sg00/Tensorizer/FlattenMacroLoop]: Enter time region
05/03/2023 07:32:24 PM INFO 29116 [FlattenMacroLoop]: Finished (changed=False)
05/03/2023 07:32:24 PM INFO 29116 [sg00/Tensorizer/FlattenMacroLoop]: Exit time region: delta=0.196s
05/03/2023 07:32:24 PM INFO 29116 [sg00/Tensorizer/SimplifyPredicates]: Enter time region
05/03/2023 07:32:26 PM INFO 29116 [SimplifyPredicates]: Finished (changed=True #instances=567387)
05/03/2023 07:32:26 PM INFO 29116 [sg00/Tensorizer/SimplifyPredicates]: Exit time region: delta=1.367s
05/03/2023 07:32:26 PM INFO 29116 [sg00/Tensorizer/InferInitValue]: Enter time region
05/03/2023 07:32:27 PM INFO 29116 [sg00/Tensorizer/InferInitValue]: Exit time region: delta=1.280s
05/03/2023 07:32:27 PM INFO 29116 [Statistics]: --- Penguin Statistics ---
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  AffinePredicateResolution  Number of const tensor evaluated from predicate values
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  AutoCastTCInputs  Tensor shrink for matmult down-cast (in bytes
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  BroadcastWeights  Number of matmults inserted to broadcast weights
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  CommuteConcat     Number of concat operand commuted
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                2  CommuteConcat     Number of concat operand distributed
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  ConcatLoopFusion  Number of concat loops fused
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  DTypeMutator      Number of bytes statically casted in weight tensor
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  DTypeMutator      Number of load whose tensor type mutated
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  DTypeMutator      Number of operator whose dst type is mutated for legalization
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                4  DataLocalityOpt   Number of deferred flush inserted for DMA write
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:              148  DataLocalityOpt   Number of delinearized weight by rewriting
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  DataLocalityOpt   Number of prefetch failed due to isl codegen
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:              277  DataLocalityOpt   Number of prefetch inserted
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  DataLocalityOpt   Number of tranposes skipped because of illegal partition address
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  DataLocalityOpt   Number of transpose inserted for DMA write
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  DeConcat          Number of concat operand distributed
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:        353452032  DeadStoreElimination  Number of bytes eliminated
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  DeadStoreElimination  Number of iteration eliminated
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  DeadStoreElimination  Number of loops eliminated
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  DeadStoreElimination  Number of strided axes
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  DeadStoreElimination  Number of tensors eliminated
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  DelinearizationBase  Number of loops tiled by delinearization
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:              118  DelinearizationBase  Number of tensors delinearized
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  DotTransform      Number of times a transformation rollback due to transformation crash
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  EliminateDivs     Number of axes tiled to eliminate div/mod
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  EliminateDivs     Number of div/mod with offset
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  ExpandISAMacro    Number of per-partition bytes copy for tensor-tensor legalization
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  ExpandISAMacro    Number of sbatomload/sbatomstore expand
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  ExpandISAMacro    Number of total per-partition bytes copy/cast for legalization
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:              402  FlattenMacroLoop  Number of axes coalesced
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:              232  FlattenMacroLoop  Number of tensors reshaped
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  GlobalLayoutOpt   Number of BFTransposes inserted
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  GlobalLayoutOpt   Number of PFTransposes inserted
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  GlobalLayoutOpt   Number of PFTransposes inserted for IO tensors
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  GlobalLayoutOpt   Number of dmas needed for transpose
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  InferNonLocalTensor  num of non-local tensors infered
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  InferPSumTensor   Number of bytes copied to psum per-partition
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  InferPSumTensor   Number of instructions whose result inferred into PSUM
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  InferPSumTensor   Number of psum copy instructions inserted
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  InferPSumTensor   Number of total bytes inferred into PSUM
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  InferStaticWeight  Number of bytes allocated as static allocation
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:              106  InferTongaTensor  Number of local tensor inferred
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  InferTongaTensor  Number of tensors that can not be inferred as local because of the non-compatible access pattern
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  InferTongaTensor  Number of tensors that cannot be inferred as local because their size cannot fit in
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  InferTongaTensor  Total Number of bytes of non-compatible tensors
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  InferTongaTensor  Total Number of bytes of tensors with large tile size
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:       2532333568  InferTongaTensor  Total number of bytes of local tensors
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  IntegerSetWrapper  Number of times compute_quota exceed
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:              200  IslSimplifier     Number of iteration eliminated
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  IslSimplifier     Number of loops eliminated
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:               88  IslSimplifier     Number of predicate eliminated
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  IslSimplifier     Number of strided axes
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  LayoutAnalysis    Number of loopnests (DAGs) whose tensors layout are not propagated but re-calculated
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  LegalizeTongaAccess  Number of per-partition bytes copy for Activation legalization
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  LegalizeTongaAccess  Number of per-partition bytes copy for MatMul legalization
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  LegalizeTongaAccess  Number of per-partition bytes copy for SBAtomLoad SB address legalization
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  LegalizeTongaAccess  Number of per-partition bytes copy for SBAtomLoad legalization
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  LegalizeTongaAccess  Number of per-partition bytes copy for SBAtomSave SB address legalization
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  LegalizeTongaAccess  Number of per-partition bytes copy for SBAtomSave legalization
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  LegalizeTongaAccess  Number of per-partition bytes copy for TensorScalarPtr legalization
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  LegalizeTongaAccess  Number of per-partition bytes copy for TensorTensor legalization
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  LegalizeTongaAccess  Number of tensor reshaped to legalize the ptr operand of the tensorscalarptr
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  LegalizeTongaAccess  Number of total per-partition bytes copy/cast for legalization
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  LegalizeTongaMacro  Explicit psum copy because of too many partitions
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  LegalizeTongaType  Number of bytes copy/cast at the producer instructions
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  LegalizeTongaType  Number of bytes copy/cast for legalization
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  LinearScanAllocator  Number of bytes allocated in psumbuffer
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  LinearScanAllocator  Number of bytes allocated in statebuffer
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  LinearScanAllocator  PSumbuffer utilization
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  LinearScanAllocator  Statebuffer utilization
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:              563  LoopFusion        Number of loops fused
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                4  LoopFusion        Number of trivial copy eliminated
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  LowerPartitionTile  Number of broadcast buffer shrink
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  LowerPartitionTile  Number of matmult instruction lowered
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  LowerPartitionTile  Number of tonga instruction (except matmult) lowered
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  LowerTranspose    Number of lossless transpose generated
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  LowerTranspose    Number of lossless transpose with bitcast generated
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  LowerTranspose    Number of lossy transpose generated
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  MPDLoopFusion     Missed loop fusions
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  MatMultCombiner   Number of matmul fused among contract axes
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  MatMultCombiner   Number of matmul fused among rows
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  MatMultCombiner   Number of matmul operand copy
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  MatMultCombiner   Number of matmul operand extend inplace
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  MatMultCombiner   Number of matmul operand extend inplace
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  MatMultCombiner   Number of weight fused for matmul fusion
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:        371048448  MemcpyElimination  Number of bytes copied by memcpy
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:        371048448  MemcpyElimination  Number of bytes eliminated
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  MemcpyElimination  Number of cast inserted
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  MemcpyElimination  Number of failure in access folding
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                5  MemcpyElimination  Number of memcopy eliminated
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  MemcpyElimination  Number of quasi affine expr generated during addr rewrite
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  MemcpyElimination  Number of simple dag moved
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  MemoryProfiler    Estimated peak psum utilization
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  MemoryProfiler    Estimated peak statebuffer utilization
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:              203  ModDivDelinear    Number of loops tiled by delinearization
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  ModDivDelinear    Number of tensors delinearized
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  ModuloAllocation  Number of dma for prefetch buffer moved
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  ModuloAllocation  Number of parallel loops generated by modulo buffer allocation
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  PadElimination    Number of padding eliminated
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  PartialLoopFusion  Number of instruction rematerialized
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  PartialLoopFusion  Number of loops fused
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  PartialLoopFusion  Number of loops fused with loop-carried dependencies
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  PolyhedralInterfereGraph  Number of inplace replacement pairs identified
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  PreScheduler      Number of instruction moved to shorten live-interval
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  PrefetchAllocation  Number of inputs/weights prefetched
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  RecognizeOpIdiom  Number of channel-wise prelu inferred
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  RecognizeOpIdiom  Number of gelu (tanh approximated) inferred
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  RecognizeOpIdiom  Number of gelu inferred
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  RecognizeOpIdiom  Number of mish inferred
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  RecognizeOpIdiom  Number of prelu inferred
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  RecognizeOpIdiom  Number of rsqrt inferred
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  RecognizeOpIdiom  Number of softplus inferred
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  Recompute         Number of bytes recomputed
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  Recompute         Number of flops recomputed
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  RelaxPredicates   Number of predicates dropped
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:     413576724480  RelayFE           Number of MAC count in relay
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:      32592370892  RelayFE           Number of total read bytes of FMAP and weight in relay
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:              269  ReshapeWeights    Number of weights being reshaped into 2-D array
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  ResolveAccessConflict  Number of dag splitted to resolve access conflict
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  ResolvePredicates  Number of subdomain generated
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  RetileSIMDMacro   Number of SIMD macro retiled to match size of psum
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                2  RewriteReplicationMatmul  Number of replication access generated for matmul
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  RewriteReplicationMatmul  Number of software replication generated
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  SimplifyTensorBase  Number of dead stores deleted
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  SimplifyTensorBase  Number of tensors simplified
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  SimplifyTensorBase  Number of tensors splitted
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  SimplifyTensorBase  Number of tensors squeezed
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  SpillPSum         Number of total per-partition bytes spilled from psum to sb
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  SplitAccGrp       Number of accumulation groups splitted
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  SplitFreeDim      Number of tensors whose blocks are further split
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  SplitUnionSets    Number of subdomain generated
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  StaticProfiler    Arithmetic intensity
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  StaticProfiler    Arithmetic intensity upper bound if we only load/store I/O tensors once (excluding gather weight tensor)
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  StaticProfiler    Arithmetic intensity upper bound if we only load/store I/O tensors once (excluding weights)
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  StaticProfiler    Arithmetic intensity without non-local tensor loads
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  StaticProfiler    Arithmetic intensity without weights and non-local loads
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  StaticProfiler    Average dma length per-partition
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  StaticProfiler    Average dma length per-partition (exclude weights)
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  StaticProfiler    Average partition utilization
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  StaticProfiler    Average pe (systolic array) utilization
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  StaticProfiler    Biggest tensor tile size in StateBuffer/PSum
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  StaticProfiler    Matmul compute efficiency: (Relay MAC counts/128/64/2) / (Total elements per partition)
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  StaticProfiler    Num of matmul transpose instructions
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  StaticProfiler    Number of arithmetic computation
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  StaticProfiler    Number of arithmetic computation by activation
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  StaticProfiler    Number of arithmetic computation by gettensorptr
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  StaticProfiler    Number of arithmetic computation by matmul
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  StaticProfiler    Number of arithmetic computation by reciprocal
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  StaticProfiler    Number of arithmetic computation by tensorreduce
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  StaticProfiler    Number of arithmetic computation by tensorscalar
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  StaticProfiler    Number of arithmetic computation by tensorscalarptr
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  StaticProfiler    Number of arithmetic computation by tensortensor
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  StaticProfiler    Number of bytes moved by tensor copy and shuffle
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  StaticProfiler    Number of bytes of weights loaded
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  StaticProfiler    Number of bytes of weights loaded in partition 0
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  StaticProfiler    Number of bytes transfer between ddr and local memory
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  StaticProfiler    Number of dma whose length is smaller than 512 bytes
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  StaticProfiler    Number of dma whose length is smaller than 512 bytes (exclude weights)
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  StaticProfiler    Number of matmul instructions
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  StaticProfiler    Number of tensorcopy after transpose
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  StaticProfiler    Number of tensorcopy from psum
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  StaticProfiler    Number of tensorcopy instructions
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  StaticProfiler    Number of tensorcopy to psum
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  StaticProfiler    Number of total dynamic instances of tonga instructions (number of instructions after unroll)
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  StaticProfiler    Utilized psum size / Max psum size
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  StaticProfiler    io tensor size / ddr transfer size
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  StaticProfiler    io tensor size / ddr transfer size ignoring nonlocal/gather tensors
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  TensorInitialization  Number of explicit initialization inserted
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  TilingForNewLayout  Number of inherit tiles
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  TilingForNewLayout  Number of instructions after tiling
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  TilingForNewLayout  Number of places global layout clobbered
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:          38.7882  TilingProfiler    Average partition utilization calculated after tiling
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:          33.9699  TilingProfiler    Average pe (systolic array) utilization calculated after tiling
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  TilingProfiler    Number of insts from dmas after tiling
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  TilingProfiler    Number of insts from gather/scatter after tiling
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:           513879  TilingProfiler    Number of insts from matmults after tiling
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  TilingProfiler    Number of insts from pf transposes after tiling
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  TilingProfiler    Number of insts from pf transposes for IO tensors after tiling
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  TilingProfiler    Number of insts from pf transposes for local tensors after tiling
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  TilingProfiler    Number of insts from pf transposes for nonlocal tensors after tiling
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  TilingProfiler    Number of insts from reduce macros after tiling
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                7  TilingProfiler    Number of non-IO DRAM tensor after tiling
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  TilingProfiler    Number of pf transposes
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  TilingProfiler    Number of pf transposes for IO tensors
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  TilingProfiler    Number of pf transposes for local tensors
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  TilingProfiler    Number of pf transposes for nonlocal tensors
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:           548535  TilingProfiler    Number of total insts after tiling
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  TongaCpyElim      Number of failure in access folding
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  TongaCpyElim      Number of tensor copy/cast eliminated
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  TongaInstComb     Number of bias_add combined to activation
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  TongaInstComb     Number of result of boolean computation shrink
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  TongaInstComb     Number of scale combined to activation
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  TongaInstComb     Number of tensorscalar fold into tensorscalarptr
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  TongaInstComb     Number of tensortensor add replace by psum accumulation
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  TongaInstComb     Number of transpose pairs folded
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  TongaInstComb     Number of trivial access pattern eliminated
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  TongaLICM         Number of instruction hoisted
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  TongaLICM         Number of instruction sunk
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  TongaLayoutGroupAnalysis  Number of loopnests (DAGs) whose tensors layout are not propagated but re-calculated
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  TongaLoopFusion   Number of loops fused
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  TongaLoopFusion   Number of matmul fused for psum locality
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:               45  TongaSizeTiling   Number of inherit tiles
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  TongaSizeTiling   Number of instructions after tiling
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                9  TongaSizeTiling   Number of places global layout clobbered
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  TongaValueNumbering  Number of instructions deleted
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  TongaValueNumbering  Number of tensors deleted
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  TransformLayoutPass  Number of static layout transforms on input/outputs
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  TransformLayoutPass  Number of transpose inserted
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  TransformLayoutPass  Number of transpose insertions failed because of predicates
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  TransformLayoutPass  Number of unknown transpose inserted
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  VDNSPrep          Number of tensor rewritten for virtual node splitting
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:               28  ValueNumbering    Number of instructions deleted
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:               28  ValueNumbering    Number of tensors deleted
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  Vectorizer        Number of instruction vectorized
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  WeightCoalescing  Number of empty loopnest eliminated
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  WeightCoalescing  Number of load instruction merged
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:        161997382  WeightRewriter    Number of bytes re-written for weights
05/03/2023 07:32:27 PM INFO 29116 [Statistics]:                0  WeightRewriter    Number of bytes reused for weight rewrite
05/03/2023 07:32:28 PM INFO 29116 [root/Tensorizer/All]: Exit time region: delta=29.905s
```

Thank you!

**Comment by aws-taylor** [comment]
Hello @andreadasilvabaudet,

 We believe we have fixed this issue in a recent release. Can you attempt to upgrade your AWS Neuron stack and try again. We have also taken measures to improve the error message around this code and will continue to do so in future release.
 
 Regards,
 Taylor

**Comment by mrnikwaws** [comment]
No response from the customer since May - as recommended above, please reinstall torch_neuron and retry.  

Please log a new issue if the issue still exists.
